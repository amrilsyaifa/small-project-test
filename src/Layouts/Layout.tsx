import React from 'react';
import { Layout as AntdLayout } from 'antd';
import 'antd/dist/antd.min.css';
import Styles from './Layout.module.scss';
import PageHeader from 'src/Reusables/Components/PageHeader';

const { Content } = AntdLayout;

interface ILayout {
  children: React.ReactNode;
}

const Layout: React.FC<ILayout> = ({ children }) => {
  return (
    <div className={Styles['layout-content']}>
      <AntdLayout>
        <div className={Styles['page-header-container']}>
          <div className={Styles['page-header']}>
            <PageHeader />
          </div>
        </div>
        <div className={Styles['container']}>
          <Content style={{ padding: '0 50px' }}>
            <div className={Styles['site-layout-content']}>{children}</div>
          </Content>
        </div>
      </AntdLayout>
    </div>
  );
};

export default Layout;
