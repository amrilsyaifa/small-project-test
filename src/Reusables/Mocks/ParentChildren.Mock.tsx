import React from 'react';

/**
 * Separate children from Component so it's not circular referencing when stringified
 */
interface IParentChildrenSeparator {
  mockLabel: string;
  children: React.ReactNode;
  MockElement?: React.FC;
}
export const ParentChildrenSeparator: React.FC<IParentChildrenSeparator> = ({
  mockLabel,
  children,
  MockElement,
  ...props
}) => {
  const RenderElement = MockElement ? MockElement : MockDivElement;

  return (
    <RenderElement data-testid={mockLabel} data-props={JSON.stringify(props)}>
      {children}
    </RenderElement>
  );
};

export const MockDivElement = ({ ...props }) => <div {...props} />;

export default ParentChildrenSeparator;
