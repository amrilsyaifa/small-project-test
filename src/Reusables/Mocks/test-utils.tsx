import React from 'react';
import { render as rtlRender } from '@testing-library/react';
import { configureStore } from '@reduxjs/toolkit';
import { Provider } from 'react-redux';
import { persistedReducer } from 'src/Store';
// Import your own reducer

interface IWrapper {
  children: React.ReactNode;
}

const render = (
  ui: React.ReactElement<any, string | React.JSXElementConstructor<any>>,
  { store = configureStore({ reducer: persistedReducer }), ...renderOptions } = {}
) => {
  const Wrapper: React.FC<IWrapper> = ({ children }) => {
    return <Provider store={store}>{children}</Provider>;
  };
  return rtlRender(ui, { wrapper: Wrapper, ...renderOptions });
};

// re-export everything
export * from '@testing-library/react';
// override render method
export { render };
