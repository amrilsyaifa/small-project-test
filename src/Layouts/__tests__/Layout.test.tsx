import { render, screen, cleanup } from 'src/Reusables/Mocks/test-utils';
import React from 'react';
import Layout from '../Layout';
import { MatchMediaObject } from 'src/Reusables/Mocks/MatchMediaObjectTest';

afterEach(cleanup);
beforeEach(() => {
  MatchMediaObject();
});

describe('Layout', () => {
  test('Layout Render Snapshoot', () => {
    const { container } = render(
      <Layout>
        <div>render layout</div>
      </Layout>
    );
    expect(container).toMatchSnapshot();
  });

  test('PageHeader Have Children render correctly', () => {
    const childrenText = 'render layout';
    render(
      <Layout>
        <div>{childrenText}</div>
      </Layout>
    );
    const titleHeader = screen.getByText(childrenText);
    expect(titleHeader).toBeInTheDocument();
  });
});
