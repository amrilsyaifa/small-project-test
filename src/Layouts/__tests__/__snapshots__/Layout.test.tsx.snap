// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout Layout Render Snapshoot 1`] = `
<div>
  <div
    class="layout-content"
  >
    <section
      class="ant-layout"
    >
      <div
        class="page-header-container"
      >
        <div
          class="page-header"
        >
          <div
            class="container"
          >
            <div
              class="header-wrapper"
            >
              <img
                alt="logo"
                src="/logo512.png"
              />
            </div>
            <div
              class="wrapper-right-text"
            >
              <div
                class="wrapper-text"
              >
                <span
                  class="ant-typography text"
                >
                  Login
                </span>
                <span
                  class="ant-typography text"
                >
                  Logout
                </span>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="container"
      >
        <main
          class="ant-layout-content"
          style="padding: 0px 50px;"
        >
          <div
            class="site-layout-content"
          >
            <div>
              render layout
            </div>
          </div>
        </main>
      </div>
    </section>
  </div>
</div>
`;
